include/zephyr/bluetooth/gatt.h

```diff
@@ -2076,6 +2076,9 @@ struct bt_gatt_subscribe_params {
  *  notified value. One may then decide whether to unsubscribe directly from
  *  this callback. Notification callback with NULL data will not be called if
  *  subscription was removed by this method.
+ *  If a different user/application has already subscribed to the same handle,
+ *  then the callback will contain the values aggregated of all the
+ *  users/applications.
```

Comment:
Would it make sense to have an example of this? 

There are 2 callbacks involved here. 
One is the `bt_gatt_subscribe_params.notify` which contains the data as well as the params when we receive a notification or indication. The callback itself does not tell the application if it received an indication or notification (which is probably should have). 

The other is the `bt_gatt_subscribe_params.subscribe` which is basically the write response. This just gives back the params. 

So none of the callbacks actually contain information about whether it was a notification or indication. 

But the documentation you've added here is wrong since the callback does not contain information about the type. What can happen is that App A subscribes to notifications and App B subscribes to indications, and looking at `att_notify`, `att_indicate` and `bt_gatt_notification`, these are just handled in the same way. So even if App A only subscribes to notifications, it may receive indications, or vice versa. 

This is what the documentation should say, unless we want to take it a step further and change the behavior so that App A only gets notifications and App B only gets indications (so that the callback behavior matches the params->value). The change should be fairly trivial.

@jhedberg @alwa-nordic what are you thoughts about this? 

And sorry that this is becoming a bigger thing that you probably intended @Yagoor :D

Reply:
I'll hold on any change until this is settled :)

Reply:
The proposal from @Thalley sounds reasonable to me

Reply:
> change the behavior so that App A only gets notifications and App B only gets indications (so that the callback behavior matches the params->value). The change should be fairly trivial.

It feel to me like we are just making random changes to our API without any use cases in mind. If it was true that notifications and indications were truly independent, this would be an obvious change. But the spec is being confusing again. I just want to understand what API the profiles will need.